base_classes/NXactivity.nxdl.xml

```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-stylesheet type="text/xsl" href="nxdlformat.xsl"?>
+<!--
+# NeXus - Neutron and X-ray Common Data Format
+# 
+# Copyright (C) 2014-2024 NeXus International Advisory Committee (NIAC)
+# 
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+# For further information, see http://www.nexusformat.org
+-->
+<definition xmlns="http://definition.nexusformat.org/nxdl/3.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" category="base" type="group" name="NXactivity" extends="NXobject" xsi:schemaLocation="http://definition.nexusformat.org/nxdl/3.1 ../nxdl.xsd">
+    <doc>
+         A planned or unplanned action that has a temporal extension and for some time depends on some entity.
+         
+         This class is a super class for all other activities.
```

Comment:
It strikes me that the intent is to capture provenance information; i.e., some DAG of actions leading up to the current state.

If so, you might like to review some existing work in recording provenance; e.g., [W3C's PROV Ontology](https://www.w3.org/TR/prov-o/)

Reply:
Also, the document says "This class is a super class for all other activities."

However, I believe this is only true if some other class extends NXactivity, which doesn't happen.  Instead NXhistory includes a group of type NXactivity, rather than extending from it.

Perhaps better expressed as "This class is the basis for recording all activities." or something similar.